### PR TITLE
Update repo-infra bazel dependency and use new gcs_upload rule

### PR DESCRIPTION
**Pull request by ixdy**
This PR provides similar functionality to push-build.sh entirely within Bazel rules (though it relies on gsutil).

It's an alternative to #44306.

Depends on https://github.com/kubernetes/repo-infra/pull/13.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -6,8 +6,8 @@ git_repository(
 
 git_repository(
     name = "io_kubernetes_build",
-    commit = "685f15b90b454af3086ab071fdea1b6db213d1fb",
-    remote = "https://github.com/kubernetes/repo-infra.git",
+    commit = "9e85ee19cc231c66a250e7df013b3c7f6e3bbd2d",
+    remote = "https://github.com/ixdy/kubernetes-repo-infra.git",
```
Comment (ixdy) [inline WORKSPACE]: revert before merging

**Comment by ixdy** [comment]
e.g. I can now do
```console
bazel run //:ci-artifacts gs://kubernetes-release-{BUILD_USER}/bazel/{STABLE_BUILD_SCM_REVISION}
```

and it basically works as expected.

**Comment by mikedanese** [comment]
LGTM once this is pointing to repo infra

File: WORKSPACE [hunk 2]
```diff
@@ -6,13 +6,13 @@ git_repository(
 
 git_repository(
     name = "io_kubernetes_build",
-    commit = "685f15b90b454af3086ab071fdea1b6db213d1fb",
-    remote = "https://github.com/kubernetes/repo-infra.git",
+    commit = "da482220efd426e61a1339e5e81cb243d2cb1b44",
+    remote = "https://github.com/ixdy/kubernetes-repo-infra.git",
```
Comment (spxtr) [inline WORKSPACE]: Merged, please fix this.

**Comment by ixdy** [comment]
oops, I realized I should probably set a message on the md5/sha1 genrules for prettier output. testing that real quick...

File: WORKSPACE [hunk 3]
```diff
@@ -6,13 +6,13 @@ git_repository(
...
-    remote = "https://github.com/kubernetes/repo-infra.git",
+    commit = "cbed6ceeb4d4ddda0a3de42ed89649b9e41baad7",
+    remote = "https://github.com/ixdy/kubernetes-repo-infra.git",
 )
 
 git_repository(
     name = "io_bazel",
-    commit = "3b29803eb528ff525c7024190ffbf4b08c598cf2",
+    commit = "1fe52dd4b2d77a740648bc1509b68acae49deffc",
     remote = "https://github.com/ixdy/bazel.git",
```
Comment (mikedanese) [inline WORKSPACE]: what change in this repo and when can we stop using it?

Comment (ixdy) [inline WORKSPACE]: This is https://github.com/ixdy/bazel/tree/docker-tags.

I updated this to strip the newline from the file when reading the tag: https://github.com/ixdy/bazel/commit/1fe52dd4b2d77a740648bc1509b68acae49deffc

I'd been trying to get this change upstreamed in https://bazel-review.googlesource.com/#/c/8370/, but I forgot to follow up on the comments. I should rebase my change on the new docker rules now anyway.

Comment (mikedanese) [inline WORKSPACE]: the rules have since moved: https://github.com/bazelbuild/rules_docker

Comment (ixdy) [inline WORKSPACE]: right, that's what I mean by "rebase on the new docker rules" :)

**Comment by ixdy** [comment]
This PR should be good to go now.

**Comment by ixdy** [comment]
Bumped our rules_go dependency because it'd be nice to stop including the .a files in our release.

**Comment by mikedanese** [comment]
@k8s-bot unit test this

**Comment by pipejakob** [comment]
It looks like this commit caused the `ci-artifacts` target to stop building .debs, which has now broken the kubeadm e2e tests that depend on them to function.

Example failure: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-gce/1822

CI build artifacts: `gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/`

    $ gsutil ls gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-client-linux-amd64.tar.gz
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-client-linux-amd64.tar.gz.md5
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-client-linux-amd64.tar.gz.sha1
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-manifests.tar.gz
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-manifests.tar.gz.md5
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-manifests.tar.gz.sha1
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-node-linux-amd64.tar.gz
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-node-linux-amd64.tar.gz.md5
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-node-linux-amd64.tar.gz.sha1
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-salt.tar.gz
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-salt.tar.gz.md5
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-salt.tar.gz.sha1
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-server-linux-amd64.tar.gz
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-server-linux-amd64.tar.gz.md5
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-server-linux-amd64.tar.gz.sha1
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-src.tar.gz
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-src.tar.gz.md5
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-src.tar.gz.sha1
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-test.tar.gz
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-test.tar.gz.md5
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes-test.tar.gz.sha1
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes.tar.gz
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes.tar.gz.md5
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/kubernetes.tar.gz.sha1
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/bin/
    gs://kubernetes-release-dev/bazel/v1.7.0-alpha.2.466+2315008ea6cb88/extra/

(debs used to be in `build/debs`, but are missing)

**Pull request merged**